integration-cli/docker_cli_external_volume_driver_unix_test.go

```diff
@@ -103,10 +103,8 @@ func newVolumePlugin(c *check.C, name string) *volumePlugin {
 	read := func(b io.ReadCloser) (pluginRequest, error) {
 		defer b.Close()
 		var pr pluginRequest
-		if err := json.NewDecoder(b).Decode(&pr); err != nil {
-			return pr, err
-		}
-		return pr, nil
+		err := json.NewDecoder(b).Decode(&pr)
+		return pr, err
```

Comment:
if you really want to do the work for the compiler you can make it:
`return pr, json.NewDecoder(b).Decode(&pr)`  :-)